### ISSUE TITLE
/////// leftovers in HTML code when mixed with JSX

### ISSUE THREAD

**Issue by krisdante** [nippur72/ifdef-loader#44]
I have the following in the code:

```
      <Head>
/// #if PRODUCTION
        <script
          async
          src="https://www.googletagmanager.com/gtag/js?id=UA-3500050-6"
        />
/// #endif
      </Head>
```

ifdef works properly, but what happens in addition is that <head> section of the HTML markup contains multiple 
///////////////// ///////////////////////////// //////////////////

Once I remove the ifdef from JS messy characters in the head are also gone. 

I wonder at which point this problem is happening

This is part of Next.js project. with SSR

**Comment by krisdante** [comment]
OK this comes from the blank code method. where the code is replaced with //. This works fine for JS, but is not working well for JSX. This is most likely used to keep the same number of lines.

**Comment by krisdante** [comment]
Or not :). I removed the blank code generator and shashes are still there :(

**Comment by nippur72** [comment]
JSX code removing via comments would require that you "exit" from JSX e.g.
```
<div>
{
///
///<Commented_jsx></Commented_jsx>
///
}
</div>
```
I don't know how this can be done in `#ifdef-loader`, since the parser isn't aware whether you are in a JSX block or not.

**Comment by krisdante** [comment]
But the parser is run before and shall remove everything including the ifdef tags and leave nothing behind.

**Comment by nippur72** [comment]
the code isn't simply removed but it's filled with `///` to allow source maps to point to the original code. I don't remember why I didn't just use blank spaces instead... I vaguely remember there was a reason.

**Comment by nippur72** [comment]
I addressed this in `2.2.0`, now there is a new option `fill-with-banks` to be set in the configuration object. When `true` it fills with blanks (space characters) instead of `//`. By default it's `false` so it's backward compatible.

It should work with JSX as spaces are ignored. 

Please let me know if that fixes your issue.

**Comment by krisdante** [comment]
I saw the code. Looks good. Thank you for that.

I spotted that there is still some issue with source maps generated by webpack. For some reason, they get a lot of slashes picked probably from ifdfs. Bit hard to debug, but when I disabled source maps //// are gone.

I am not sure yet how it is possible that a source maps age getting the source with //// (I patched the offloader before to replace with spaces already)
